CoreImportExportPlugins/src/au/gov/asd/tac/constellation/plugins/importexport/jdbc/JDBCImportController.java

```diff
@@ -131,32 +136,36 @@ public synchronized void newActiveGraph(final Graph graph) {
                     .set(ImportJDBCPlugin.SCHEMA_PARAMETER_ID, schema)
                     .executeWriteLater(importGraph);
         }
-        return Collections.emptyList();
     }
 
     @Override
     protected void updateSampleData() {
-        final String queryCopy = this.query;
-        if (connection == null || StringUtils.isBlank(queryCopy)) {
+        final StringBuilder previewQuery = new StringBuilder(query);
+
+        // limit the preview query for efficiency
+        if (!query.toLowerCase(Locale.ENGLISH).contains(QUERY_LIMIT_TEXT)
+                && query.toLowerCase(Locale.ENGLISH).startsWith(SELECT_TEXT)) {
+            previewQuery.append(QUERY_LIMIT_TEXT).append(PREVIEW_ROW_LIMIT);
+        }
```

Comment:
This was the actual fix for the bug. Everything else was well...my OCD :laughing: